### PR TITLE
add wip personal notes on testing

### PR DESCRIPTION
**Pull request by cblgh**
I'm documenting various things I learn about the testing setup, mostly for my own sake. I figured this could be good to add to the repo once we move on from go-ssb-room. This PR tracks those changes (somewhat haphazardly, though)

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+opposition to using an entire pre-filled sqlite database of fake data.
+
+Thus, there is no command you run first to generate your fake database, but
+functions you have to call in a kind of pre-test setup, inside each testing
+block you are authoring. 
+
+> [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter) generates a bunch of methods for each function, so you have
+> XXXXReturns,  XXXCallCount XXXArgsForCall(i) etc
+>
+> _cryptix_
```
Comment (cryptix) [inline docs/testing.md:63]: This sections feels sub-optimal specific. Here are some more examples:

https://github.com/ssb-ngi-pointer/go-ssb-room/blob/fbdba34a1dfe0ea1613a47d80220cc3fdf2736b8/web/handlers/admin/allow_list_test.go#L113-L118



Having the `List()` function return a static list of three items.

https://github.com/ssb-ngi-pointer/go-ssb-room/blob/fbdba34a1dfe0ea1613a47d80220cc3fdf2736b8/web/handlers/admin/allow_list_test.go#L210-L212


Checking how often `RemoveID` was called and with what arguments.

**Comment by cryptix** [comment]
Feel free to merge this if you dont want to make more changes.

Having it is much better then not having it, IMHO.

**Comment by cblgh** [comment]
I'll definitely merge this! My thought is I'll keep it open and revise a bit, and then (I/someone) can merge it once I have pushed the last changes and we move on from go-ssb-room to other pastures

**Comment by cblgh** [comment]
@cryptix alright figured i'd just do a quick pass on what I already had, and simplify it a bit. Think this is alright to merge?

**Comment by cryptix** [comment]
Looks good!

If you have energy for it, the muxrpc stuff is missing but it can be a follow up. Feel free to merge.

**Comment by cblgh** [comment]
energy levels today 
image

**Pull request merged**